src/src/MainArea/MainArea.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react'
+import AddBookForm from './AddBookForm/AddBookForm'
+import Bookshelf from './Bookshelf/Bookshelf'
+import Archive from './Archive/Archive'
+
+interface IProps {
+    visiblePage: string;
+}
+
+type eVisibleComponent = {
+    bookShelf: JSX.Element;
+    addBookForm: JSX.Element;
+    archive: JSX.Element;
+  }
+
+const MainArea = ({visiblePage}: IProps) => {
```

Comment:
That's a very nice solution! 🙂
You can develop it further to show the visible component according to the browser address.
Those pieces of code are called routers, you can find some very impressive solutions that people are using like [React Router](https://github.com/ReactTraining/react-router)